src/python/pants/backend/codegen/protobuf/tailor.py

```diff
@@ -22,15 +22,15 @@
 
 
 @dataclass(frozen=True)
-class PutativeProtobufTargetsRequest:
+class PutativeProtobufTargetsRequest(PutativeTargetsRequest):
     pass
 
 
 @rule(level=LogLevel.DEBUG, desc="Determine candidate Protobuf targets to create")
 async def find_putative_targets(
-    _: PutativeProtobufTargetsRequest, all_owned_sources: AllOwnedSources
+    req: PutativeProtobufTargetsRequest, all_owned_sources: AllOwnedSources
 ) -> PutativeTargets:
-    all_proto_files = await Get(Paths, PathGlobs(["**/*.proto"]))
+    all_proto_files = await Get(Paths, PathGlobs, req.search_paths.path_globs("*.proto"))
```

Comment:
Should the file collection move into the `tailor` goal itself, and have the file name(s) to match be a class property of the `PutativeTargetsRequest` type? The "I expect that you will want to match files, so here's an API to help do that" is a bit roundabout if we think that all `tailor` implementations will do roughly the same thing.

Reply:
That might be premature? There might be specialized logic, although I guess we can always leave an escape hatch for that. In any case, would prefer to do that in a followup.